backend_site/app.py

```diff
@@ -0,0 +1,49 @@
+"""
+The app module, containing the app factory function.
+Factory functions allow us to have multiple configurations of the server
+(testing, production, etc.)
+See http://flask.pocoo.org/docs/1.0/patterns/appfactories/ for more info
+"""
+
+from flask import Flask
+
+from backend_site.extensions import cors
+from backend_site.settings import ProdConfig
+
+from backend_site.events import blueprint as events_routes
+from backend_site.officers import blueprint as officers_routes
+from backend_site.posts import blueprint as posts_routes
+from .views import blueprint as static_views
+
+
+def createApp(config_object=ProdConfig):
+    """
+    Factory method to create new Flask app from the modules in each subfolder
+    """
+    print(__name__)
+    print(__name__.split('.')[0])
+    app = Flask(__name__.split('.')[0],
+                static_folder=None,
+                root_path=config_object.PROJECT_ROOT)
+    app.config.from_object(config_object)
+
+    origins = app.config.get('CORS_ORIGIN_WHITELIST', '*')
```

Comment:
If we're intending to open up the API in the future, I think this should be changed.

Reply:
Changed how? Except that I forget to add in lug.cs.uic.edu to the whitelist, I'll fix that.

Reply:
Since there's a `CORS_ORIGIN_WHITELIST`, doesn't it mean that only sites inside that list can access the API (meaning that if someone wanted to use our API, they'd have to ask us to add them to said whitelist), or am I misunderstanding it here?

I guess something else to also consider is how open we plan to make the API (e.g. without keys, with keys provided by us, etc.).

Reply:
That's a good point, I didn't think of that. To be clear, whitelisting like this would prevent the API from being used in browsers by sites not located at addresses on the list, it wouldn't do anything to stop it being used in mobile apps or server-side code. Still, we should discuss this, although until we have that discussion I vote we keep it locked down.

Reply:
I think that's a good idea, I doubt too many people will want/need to use it anyway. But for those who do wish to use it we can add them on to the approved list on a case by case basis